src/dmtest/gendatablocks.py

```diff
@@ -346,6 +348,9 @@ def trim(self):
         will behave properly.
 
         """
+        byte_offset = self.block_size * self.offset
+        byte_size = self.block_size * self.block_count
+        process.run(f"blkdiscard -o {byte_offset} -l {byte_size} {self.path}")
```

Comment:
I could add an fsync option, since blkdiscard doesn't appear to do that. The default for the write method is not to fsync, so I think that makes sense for trim too. (The write method also has options for using O_SYNC and O_DIRECT but using blkdiscard doesn't give us those options.)

Either way, we do know the discards have been _acknowledged_ by the time the program returns, much like writes without O_DIRECT or O_SYNC...

Reply:
Oh, I just noticed I didn't update the doc string for this function